virtcontainers/device/drivers/block.go

```diff
@@ -67,13 +67,28 @@ func (device *BlockDevice) Attach(devReceiver api.DeviceReceiver) (err error) {
 		Index:  index,
 	}
 
-	driveName, err := utils.GetVirtDriveName(index)
-	if err != nil {
-		return err
-	}
-
 	customOptions := device.DeviceInfo.DriverOptions
-	if customOptions != nil && customOptions["block-driver"] == "virtio-blk" {
+	if customOptions != nil {
+		var globalIdx int
+
+		switch customOptions["block-driver"] {
+		case "virtio-blk":
+			globalIdx = index
+		case "virtio-mmio":
+			//With firecracker the rootfs for the VM itself
+			//sits at /dev/vda and consumes the first index.
+			//Longer term block based VM rootfs should be added
+			//as a regular block device which eliminates the
+			//offset.
+			//https://github.com/kata-containers/runtime/issues/1061
+			globalIdx = index + 1
+		}
+
+		driveName, err := utils.GetVirtDriveName(globalIdx)
```

Comment:
@amshinde I'm wondering if we still rely on `GetVirtDriveName` for `virtio-blk` device identification? I thought we now use pci address for them. Then at least `GetVirtDriveName` is not needed for `virtio-blk`. We can still keep it for `virtio-mmio`. In future we might want to replace it with disk UUID though.

Reply:
Agree with @bergwolf 

Overall it LGTM